client/securedrop_client/gui/widgets.py

```diff
@@ -690,20 +857,94 @@ def show_sources(self, sources: list[Source]) -> None:
                 # Then call the function to remove the wrapper and its children.
                 self.delete_conversation(source_uuid)
 
+        # Show the correct conversation pane gui element depending on
+        # the number of sources a) available and b) selected.
+        # An improved approach will be to create an `on_source_context_update`
+        # pyQtSlot that subscribes/listens for storage updates and calls
+        # `show_sources` and `show_conversation_context`.
+        self._on_update_conversation_context()
+
+    def _on_update_conversation_context(self) -> None:
+        """
+        Show the correct view type based on the number of available and selected sources.
```

Comment:
This is great.  It really shows off the `QStackedLayout`.

Reply:
Thanks. As you can probably deduce, I'm foreshadowing: this should eventually be a pyqtSlot and not a manually-triggered method :)  And the goal would be to have this be a callback that accepts a "state" / status, implemented in any responsive gui component. The controller would send out one signal (`controller.emit(LOGGED_IN)`, `controller.emit(SELECTED[source1]))`, `controller.emit(SELECTED[Source1,Source2,Source3]))`, all widgets would implement the same `_on_update` method with the same signature that displays the correct view, instead of manually updating the gui components in a bunch of places.

:)